pkg/specs/apptype/determine_type.go

```diff
@@ -101,6 +104,26 @@ func (r *inspector) determineTypeFromContents(
 
 	err = fetcher.GetFiles(ctx, upstream, savePath)
 	if err != nil {
+		if _, ok := err.(errors2.FetchFilesError); ok {
+			r.ui.Info(fmt.Sprintf("Failed to retrieve upstream %s", upstream))
+
+			var retryError error
+			retries := r.viper.GetInt("retries")
+			for idx := 1; idx <= retries; idx++ {
+				debug.Log("event", "retry.getFiles", "attempt", idx)
+				r.ui.Info(fmt.Sprintf("Retrying to retrieve upstream %s ...", upstream))
+
+				time.Sleep(time.Second * 5)
+				retryError = fetcher.GetFiles(ctx, upstream, savePath)
+
+				if retryError != nil {
+					r.ui.Info(fmt.Sprintf("Retry attempt %v out of %v to fetch upstream failed", idx, retries))
+					level.Error(r.logger).Log("event", "getFiles", "err", retryError)
+				} else {
+					return "", "", nil
```

Comment:
I think this block needs some reworking
Correct flow should be return error after retries run out (including if 0 retries) and continue on into the next code on success